File: 0014-longest-common-prefix/0014-longest-common-prefix.py
class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        commonPrfix = strs[0]

        for str in strs[1:]:
            while str.find(commonPrfix) != 0:
                commonPrfix = commonPrfix[:-1]
                if not commonPrfix:
                    return ""
        
        return commonPrfix

File: 0014-longest-common-prefix/0014-longest-common-prefix.js
/**
 * @param {string[]} strs
 * @return {string}
 */
var longestCommonPrefix = function(strs) {
    let commonPrefix = strs[0]; 

    for (let i = 1; i < strs.length; i++) {
            while (strs[i].indexOf(commonPrefix) !== 0) {
            commonPrefix = commonPrefix.slice(0, -1);
            if (!commonPrefix) return "";
        }
    }

    return commonPrefix;
};